cypress/integration/visual-regression/card.js

```diff
@@ -0,0 +1,56 @@
+describe(`Card Component`, () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:4200/components/card');
+  })
+  describe(`interaction`, () => {
+    it(`should support interaction when set to true`, () => {
+      cy.get(':nth-child(1) > .ng-untouched').click().then(() => {
+        cy.get('#ts-card-0 > .mat-ripple').click();
+        cy.percySnapshot('interaction');
+      })
+    });
+  });
+
+  describe(`center`, () => {
+    it(`should center text when set to true`, () => {
+      cy.get(':nth-child(2) > .ng-untouched').click().then(() => {
+        cy.get('#ts-card-0').should('have.class', 'c-card--centered');
+        cy.percySnapshot('center');
+      });
+    });
+  });
+
+  describe(`flat`, () => {
+    it(`should have flat class set when flat flag set to true`, () => {
+      cy.get(':nth-child(3) > .ng-untouched').click().then(() => {
+        cy.get('#ts-card-0').should('have.class', 'c-card--flat');
+        cy.percySnapshot('flat');
+      });
+    });
+  });
+
+  describe(`border and color`, () => {
+    it(`should set border color accordingly`, () => {
+      cy.get(':nth-child(5) > .ng-valid').select('bottom');
+      cy.get('#ts-card-0').should('have.class', 'c-card--border-bottom');
+      cy.percySnapshot('bottom-border');
+    });
+
+    it(`should set right color`, () => {
+      cy.get(':nth-child(4) > .ng-valid').select('warn');
```

Comment:
This type of selector seems fragile.. if we added another instance in place #2 would it break all these references?

Reply:
Would we be able to add IDs to test components?

Reply:
👍 yeah thought about that as todo as I was trying to get the functionality working. Those selectors are from cypress - they output a selector if you point anywhere on the page.